src/thread.rs

```diff
@@ -355,14 +357,45 @@ impl<'mir, 'tcx: 'mir> ThreadManager<'mir, 'tcx> {
         // Mark the joined thread as being joined so that we detect if other
         // threads try to join it.
         self.threads[joined_thread_id].join_status = ThreadJoinStatus::Joined;
+
+        self.wait_on_thread(joined_thread_id, None, data_race)
+    }
+
+    fn wait_on_thread(
```

Comment:
Please add a doc comment.

So the new thing is that windows supports waiting with a timeout but Linux (or at least the part of the API that we implement) does not? Do we have a test covering join with a timeout?

Reply:
I really should have added a test for this from the beginning. It turns out I really messed up the implementation :/

Reply:
Is this even accessible via stdlib APIs? If not, I think the better thing to do would be to *not* support timeouts for now, and make WaitOnAddress just `throw_unsup_format` when a timeout is specified.